lib/modules/rust-cargoaudit/index.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const path = require('path')
+const fs = require('fs')
+const ModuleResults = require('../../results')
+const exec = require('../../exec')
+const logger = require('../../logger')
+
+const key = __dirname.split(path.sep).pop()
+
+module.exports = {
+  key,
+  description: 'Checks Rust projects for dependencies with known vulnerabilities',
+  enabled: true,
+  handles: async fm => {
+    const isRustProject = fs.existsSync(path.join(fm.target, 'Cargo.toml'))
+    const hasCommand = await exec.exists('cargo')
+
+    if (isRustProject && !hasCommand) {
+      logger.warn('Cargo.lock found but cargo was not found in $PATH')
+      logger.warn(`${key} scan will not run unless you install cargo`)
+      return false
+    }
+
+    return isRustProject
+  },
+  run: async fm => {
+    let lockFileWasGeneratedByTheModule = false
+    try {
+      if (!fs.existsSync(path.join(fm.target, 'Cargo.lock'))) {
+        await exec.command('cargo generate-lockfile', { cwd: fm.target })
+        lockFileWasGeneratedByTheModule = true
+      }
```

Comment:
If the package is library then `Cargo.lock` should not be present and `cargo audit` will fail.
IMO it makes sense to generate lock file and remove it after processing to make it useful for library owners as well.
Opinions?

Reply:
I am divided on this: On the one hand, we should avoid “polluting” the target folder, on the other hand this is really useful. This might lead to some `UID` and `GID` issues when running a CI, but this can be easily mitigated with the proper `docker run` command. Unfortunately `cargo generate-lockfile` gives us no option as to *where* the lockfile should be created, otherwise I’d have put it into a temporary folder. 
I’d say we go this route until someone complains :)

Reply:
There won't be any pollution because we delete it later, but there might be an issue with access to create lock file.

> I’d say we go this route until someone complains :)

Totally agree 😄